base_layer/core/src/validation/block_validators.rs

```diff
@@ -51,12 +51,17 @@ pub const LOG_TARGET: &str = "c::val::block_validators";
 #[derive(Clone)]
 pub struct OrphanBlockValidator {
     rules: ConsensusManager,
+    bypass_range_proof_verification: bool,
```

Comment:
This can be potentially dangerous. See main response.

Reply:
I mentioned that this was dangerous in the description. It's kind of the point of this PR

Reply:
Yes, but we have 2 distinct cases for dealing with blocks. 
1: `OrphanBlockValidator`, this only deals with propagated blocks
2: `BlockValidator` deals with syncing. 

So in the sync case, if you don't check range proofs, you can get a fork and only you end up on the bad chain. 
but in the propagation case, if you don't check the range proofs you will send out this bad block to your peers and then you can get banned. 

Thats why I suggest to only do this on sync, and or disable the propagation of the block if you don't check range proofs. That way you won't get banned for a bad block.

Reply:
Yes, I agree with you it is dangerous and should not be run. 

So let's look at the use cases:
1. Fast sync
2. Lite mode clients

While enabling it only during fast sync,  the config setting is misleading and should be renamed to a specific `lite_mode` sync strategy or something like that.

If you take it out of normal operation, you prevent it's use in use case 2. For example, say you have a slow PI that is running this, but does not want to run block validation, for e.g. SPV. This is not possible.

Reply:
This is really mostly for debugging and profiling. I thought about renaming the config to `debug_bypass_range_proof_verification`, but it really is an advanced feature and should not be toggled with. Also there are other features that are debug only. E.g. `max_difficulty`, `force_sync_peers`, `pruning_horizon at 1`

Reply:
I like the idea of naming it `lite_mode` or something. But if we go by bypassing range proof verification for block propagation as well, then should we not disable block propagation on that node. So that that node only asks for blocks, sends out blocks that are asked for. It will broadcast a block it receives from the network. This makes more sense in the lite SPV type mode. An SPV-type lite node should not be relied on for network integrity and the like.

Reply:
Great you can do that in a follow up PR. This is meant as a debug only feature to measure profiling